File: C#/Practice/LeetCode/Easy/Count of Matches in Tournament.cs
// https://leetcode.com/problems/count-of-matches-in-tournament/description/

public class Solution {
    public int NumberOfMatches(int n) {
        int numberOfTeams = n;
        int matchesCounter = 0;

        while (numberOfTeams != 1) {
            matchesCounter += (int)Math.Floor((double)numberOfTeams / 2);
            numberOfTeams = (int)Math.Ceiling((double)numberOfTeams / 2);
        }
        
        return matchesCounter;
    }
}

File: JavaScript/Practice/LeetCode/Easy/Count of Matches in Tournament.js
// https://leetcode.com/problems/count-of-matches-in-tournament/description/

/**
 * @param {number} n
 * @return {number}
 */
var numberOfMatches = function(n) {
    let numberOfTeams = n;
    let matchesCounter = 0;

    while (numberOfTeams !== 1) {
        matchesCounter += Math.floor(numberOfTeams / 2);
        numberOfTeams = Math.ceil(numberOfTeams / 2);
    }
    
    return matchesCounter;
};